File: packages/sdk-android/android/everframe-core/src/main/kotlin/dev/everframe/config/ConfigValidator.kt
// SPDX-License-Identifier: MIT
// SPDX-FileCopyrightText: 2026 ScriptX
//
// Validates a EverframeConfig before Everframe.start(...) installs it. Throws
// EverframeConfigError on bad input — synchronously, so misconfiguration fails
// at app boot (not at runtime).
//
// Endpoint scheme validation is gone: the ingest URL is no longer a public
// config field. It's baked into BuildConfig.INGEST_URL per-variant (release
// = prod, debug = EVERFRAME_DEV_INGEST_URL env). See IngestEndpoint.kt.
package dev.everframe.config

object ConfigValidator {
    fun validate(config: EverframeConfig) {
        if (config.appId.isBlank()) throw EverframeConfigError.MissingAppId
        if (config.sdkKey.isBlank()) throw EverframeConfigError.BlankSdkKey
        val rate = config.vitals.sampleRate
        if (rate != null && (rate.isNaN() || rate !in 0.0..1.0)) throw EverframeConfigError.InvalidVitalsSampleRate
    }
}

File: packages/sdk-ios/Sources/Everframe/Config/ConfigValidator.swift
// SPDX-License-Identifier: MIT
// SPDX-FileCopyrightText: 2026 ScriptX
//
// Validates an EverframeConfig before Everframe.shared.start(_:) installs it.
// Throws EverframeConfigError on bad input.
//
// Endpoint scheme validation is gone: the ingest URL is no longer a public
// config field. The URL is baked at compile time via IngestEndpoint.swift
// (Release builds get https://everframe.dev; Debug builds may read the
// EVERFRAME_DEV_INGEST_URL env var).
import Foundation

enum ConfigValidator {
    static func validate(_ config: EverframeConfig) throws {
        // Per Phase 04.2 D-03: hard-fail at start() if the Everframe SDK key is missing,
        // wrong-prefix, or wrong-length. Reuses .missingAppId verbatim — the
        // appId/sdkKey rename is explicitly deferred (D-07). Length 41 = prefix
        // "txx_live_" (9) + 32-char body matching admin SdkKeysPanel emission.
        let trimmed = config.appId.trimmingCharacters(in: .whitespacesAndNewlines)
        if trimmed.isEmpty || !trimmed.hasPrefix("txx_live_") || trimmed.count != 41 {
            throw EverframeConfigError.missingAppId
        }
        // Session Vitals (iOS spec 2026-09-05 §1) — a local sample-rate
        // override outside [0,1] (including NaN) is a caller bug, not a
        // silent clamp.
        if let rate = config.vitals.sampleRate, rate.isNaN || rate < 0 || rate > 1 {
            throw EverframeConfigError.invalidVitalsSampleRate
        }
    }
}
